### PR TITLE
feat(unlock-app, schema) - enable custom text for redirect URL

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description
Enable custom text for `return CTA` from checkout Builder.

A new field is visible in the checkout builder when nothing is set the form will form as always: 

Screenshot 2023-04-06 at 12 18 29

When a custom CTA is set on the field the checkout builder, it will show on the checkout as custom text from the CTA

**Custom text on checkout builder**

Screenshot 2023-04-06 at 12 26 35

**Custom text on checkout**

Screenshot 2023-04-06 at 12 26 32

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/checkout/main/Returning.tsx [hunk 1]
```diff
@@ -191,9 +191,15 @@ export function Returning({
...
               </Button>
             ) : (
-              <div className="flex justify-between gap-4">
+              <div
+                className={`gap-4 ${
+                  paywallConfig?.endCta
+                    ? 'grid grid-cols-1'
+                    : 'flex justify-between '
+                }`}
+              >
```
Comment (kalidiagne) [inline unlock-app/src/components/interface/checkout/main/Returning.tsx:200]: As the text can be very long when a custom CTA in present lets show buttons in 2 rows

Screenshot 2023-04-06 at 12 29 53

File: packages/core/src/schema.ts [hunk 2]
```diff
@@ -150,6 +149,11 @@ export const PaywallConfig = z
           '(Optional) If supplied, the user is prompted to sign this message using their wallet. Your application needs to handle the signature to identify the user.',
       })
       .optional(),
+    endCta: z
```
Comment (julien51) [inline packages/core/src/schema.ts:152]:
```suggestion
    endCallToAction: z
```

File: packages/core/src/schema.ts [hunk 3]
```diff
@@ -150,6 +149,11 @@ export const PaywallConfig = z
...
+      .string({
+        description: 'Show a custom text for redirect URL.',
```
Comment (julien51) [inline packages/core/src/schema.ts:154]:
```suggestion
        description: 'Show a custom text on the final button that triggers a redirect',
```

File: packages/types/src/types/unlockTypes.ts [hunk 4]
```diff
@@ -174,6 +174,7 @@ export interface ChainExplorerURLBuilders {
   [site: string]: (_address: string) => string
 }
 
+// TODO: to remove, deprecated
```
Comment (julien51) [inline packages/types/src/types/unlockTypes.ts:177]: Great find! Please do that in a follow up PR?

**Review by julien51 — APPROVED** [review]
Looks good but please let's use a better name!

**Pull request merged**